### DISCUSSION TITLE
Service profile to HTTP Route filters integration

### DISCUSSION THREAD

**Discussion by dashingsat** [linkerd/linkerd2#8955] [General]
Hello Linkerd Contributors group,

We are in the process of experimenting to add in some new features to Linkerd proxy , for e.g adding a rate limiter. We were thinking to add this change as part of HTTP Route filters like https://github.com/linkerd/linkerd2-proxy/tree/main/linkerd/http-route/src/http/filter. I couldn't figure out in the code that how could this be configured through Control plane. 

For e.g having something like this
rate-limit-configuration:
   threshold_no_of_request: 100
   time_window: 1 sec
   burst_percentage: 30%

The example here is just symbolical, we want to see some existing reference of integration.

We were actually thinking to add the corresponding configuration in the Service profile. Any reference to code/snippet where the service profile information is actually integrated to form the filters would really be helpful.

Thanks and regards
Satya

**Comment by adleong** [comment]
Hi @dashingsat!  HTTPRoute filters like the ones you linked to are configured through the HTTPRoute resource, a new CRD that was recently added, and not through service profiles.  The HTTPRoute resources are read by the policy-controller (see https://github.com/linkerd/linkerd2/blob/main/policy-controller/k8s/api/src/policy/httproute.rs) and served to the proxy over the policy API (https://github.com/linkerd/linkerd2-proxy-api/blob/main/proto/inbound.proto#L20).
